sound/soc/intel/boards/sof_cs42l42.c

```diff
@@ -218,7 +218,9 @@ static int sof_audio_probe(struct platform_device *pdev)
 	dev_dbg(&pdev->dev, "sof_cs42l42_quirk = %lx\n", sof_cs42l42_quirk);
 
 	/* initialize ctx with board quirk */
-	ctx = sof_intel_board_get_ctx(&pdev->dev, sof_cs42l42_quirk);
+	ctx = sof_intel_board_get_ctx(&pdev->dev, sof_cs42l42_quirk,
+				      mach->mach_params.codec_type,
+				      mach->mach_params.amp_type);
```

Comment:
and from this code it looks like we have more parameters to shuffle around, so the code doesn't really look simpler to me...